src/Interfaces.ts

```diff
@@ -0,0 +1,47 @@
+import { ChangeEvent } from "react";
+import { AxiosResponse } from "axios";
+
+export interface AuthProps {
+  toggleLogin: Function;
```

Comment:
suggestion : can we use ```() => void``` for toggleLogin and ```React.Dispatch<React.SetStateAction<boolean>>``` for setLoading

Reply:
> Try to define return types for functions rather than calling generic Function type as this call returns an unsafe "any" return type.

Hi Saurabh, this function toggles the state and does not return anything. I think `() => void` will better justify the case.

Reply:
> suggestion : can we use `() => void` for toggleLogin and `React.Dispatch<React.SetStateAction<boolean>>` for setLoading

Hi Abhishek, yes, `() => void` will be better for `toggleLogin` and `(loading: boolean) => void` will suffice for `setLoading` as it takes a boolean and sets it as the state